### ISSUE TITLE
[ACTIVE] SWARM-CLAIM AUDIT-B|D04|CURRENT-MAIN|IDENTITY-SAFE-PRESTEP-INTEGRITY|7ACCC766|V1

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1280]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-1280
SWARM_LANE_KEY: AUDIT-B|D04|CURRENT-MAIN|IDENTITY-SAFE-PRESTEP-INTEGRITY|7ACCC766|V1
PREFERRED_LANE: #7 (D06)
PREFERRED_WORK_KIND: DATA_SOURCE_OR_PIPELINE
SELECTED_WORK_KIND: REDTEAM_AUDIT / INDEPENDENT_VERIFY / PRESTEP-INTEGRITY
PARENT_ISSUES: #5 #548 #723 #835 #856 #867 #990 #1260 #1238
STATUS: ACTIVE_PENDING_POSTCLAIM_ARBITRATION
BASE_SHA: 7accc766d76693a69dabb5ff853cb35040ec4728
CLAIMED_AT_UTC: 2026-09-11T14:35:45Z
LEASE_UNTIL_UTC: 2026-09-11T20:35:45Z

ROUTING:
Protocol formula for issue #1280 gives preferred permanent lane `2 + (1280 mod 15) = #7` and preferred work kind `work_kinds[(1280 div 15) mod 8] = DATA_SOURCE_OR_PIPELINE`. That preferred slot is saturated by current D06 Product owner #1264 and adjacent D07 #1270; opening another D06/data package would collide or increase scientific distance. Protocol marks preferred slot as a hint, not a lock, and requires rotation when saturated/blocked/low-value. This claim therefore rotates to the nearest released launch-adjacent D04 audit surface.

CANDIDATE COMPARISON:
1. Current-main D04 pre-step authority integrity — SELECTED. #1260 explicitly falsified its initial stale-ledger Product finding and released Product mutation after proving the canonical identity-safe wrapper already blocks that exact attack. A narrow audit of the actual call path can still find a *distinct* launch-relevant fail-open or positively remove D04 as a current blocker without perturbing correct code.
2. Preferred D06 lane — YIELD to active #1264; D07 to active #1270.
3. PR #1247 integration — YIELD to current integration owner #1238; exact decontam PASS_WITH_EXCLUSIONS already exists.
4. G05/G06 #1250, balance #1252/#1253, split #1246, readiness #1251, D05 #1233, pilot #1248, resource #1262 — YIELD to active owners.
5. New source family — REJECT: farther from first legal optimizer step than current post-decontam/pre-step gates.

OWNED_SURFACES:
- read-only current main exact SHA `7accc766d76693a69dabb5ff853cb35040ec4728`;
- canonical D04 path: `src/twelve_six/data/identity_safe_exposure_guard.py`, `src/twelve_six/data/deterministic_exposure_order.py`, `src/twelve_six/data/unique_loss_ledger_v2.py`, directly coupled tests/consumers only as necessary;
- verify that externally expected packed/ledger identities, closed-world schema, self-hash, one-pass capacity, replay/resume and oversubscription constraints are enforced at the actual caller boundary before any loss position can be authorized;
- audit read-only until a NEW defect distinct from #1260's falsified suspicion is proven.

AVOID_SURFACES:
- no reopening #1260's false-positive base-class repair;
- no D03 corpus/decontam/privacy/balance/split Product mutation;
- no D05/D06/D07/readiness/recipe/pilot/integration mutation;
- no corpus payload execution, tokenizer fit, optimizer/model update, final-test outcome access, paid compute, foreign pretrained weights, external LLM/API data/intelligence.

OBJECTIVE:
Determine from exact current code whether the canonical D04 launch path can authorize positive unique causal loss positions from a coherently substituted or malformed ledger/packing state despite external expected identities, or whether D04 is already fail-closed and therefore should not consume another Product repair cycle. If a distinct defect is proven, rotate this same registration to the exact same-lineage repair semantic after fresh arbitration and make the smallest repair.

PACKAGE_DIMENSIONS:
- implementation_or_primary_research: exact-code adversarial audit of launch path;
- adversarial_or_negative_tests: reproduce coherent substitution/type/replay/oversubscription paths from existing tests or executable code reasoning;
- live_authority_binding: exact current-main SHA and canonical callers;
- end_to_end_or_integration_proof: trace expected identity -> guard construction -> authorization/resume call path;
- documentation_or_operator_handoff: durable exact finding/no-new-blocker verdict to D04/readiness/pilot owners.

ACCEPTANCE:
1. postclaim direct paginated OPEN issue collection confirms no earlier live exact-semantic audit owner; #1260's released false-positive repair is not treated as active ownership of this distinct audit;
2. exact main remains `7accc766...` during verdict or audit is rebound before conclusion;
3. actual canonical callers are traced, not only the low-level base class;
4. stale identity, coherent reseal, unknown keys, bool/int aliases, segment/capacity drift, replay/resume and oversubscription are assessed at the externally expected identity boundary;
5. verdict is either a concrete reproducible NEW finding with minimal same-lineage repair guidance, or `NO_NEW_D04_BLOCKER` with exact protected path/evidence; no speculative patch;
6. no optimizer step/training/final-test/paid compute is executed by this audit.

FALLBACK:
Collision => mark ABANDONED_DUPLICATE and rotate this same #1280 registration. If the audit finds no new blocker, publish the exact no-new-blocker result and immediately rotate to the next genuinely unowned finish-first gate; do not create a cosmetic PR.

TRUTH_BOUNDARY:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
OPTIMIZER_UPDATES_EXECUTED=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

**Comment by Oleksii-debug** [comment]
POSTCLAIM DIRECT OPEN ARBITRATION — WIN

Direct GitHub OPEN issue/PR collection was exhausted newest-first with `per_page=100`: pages 1–8 nonempty, page 9 empty. No earlier live claim owns exact semantic key `AUDIT-B|D04|CURRENT-MAIN|IDENTITY-SAFE-PRESTEP-INTEGRITY|7ACCC766|V1` or an equivalent read-only audit of the actual canonical identity-safe pre-step call path at current main.

Related #1260 is NOT a collision for this audit: its authoritative checkpoint explicitly says `SCOUT FINDING FALSIFIED / PRODUCT MUTATION ABANDONED`, `STATUS: RELEASED_FALSE_POSITIVE / ROTATING`, no Product file changed and no PR opened after proving the canonical identity-safe wrapper already blocks that exact stale-ledger suspicion. SWARM-1280 will not reopen that base-class repair.

Newer/adjacent workers are disjoint: #1279 audits PR #1078; D06/D07 and the central post-decontam gates remain owned by their existing workers. Integration authority remains #1238.

STATUS: ACTIVE_ARBITRATED_READ_ONLY_AUDIT
PRODUCT_MUTATION=false
MERGE_AUTHORITY=false
TARGET_MAIN=7accc766d76693a69dabb5ff853cb35040ec4728

Proceeding now to exact-source caller tracing. Any main movement affecting owned D04 surfaces will force rebinding before verdict. A distinct proven defect will require a fresh same-registration repair arbitration before any branch/file mutation.

**Comment by Oleksii-debug** [comment]
D04 CURRENT-MAIN PRE-STEP INTEGRITY AUDIT — `NO_NEW_D04_BLOCKER` on exact main `7accc766d76693a69dabb5ff853cb35040ec4728`

Scope audited read-only after arbitration WIN:
- `src/twelve_six/data/identity_safe_exposure_guard.py`
- `src/twelve_six/data/deterministic_exposure_order.py`
- `src/twelve_six/data/ordered_exposure_resume.py`
- `src/twelve_six/data/unique_loss_ledger_v2.py`
- `tools/validate_d04_ordered_exposure_runtime.py`
- launch-input binding and current canonical carrier #802 boundary needed to distinguish corpus training from mechanics-only pilot code.

Exact result:
1. `require_expected_ledger_identity` requires an externally supplied expected ledger SHA, exact closed-world schema, and recomputed self-hash before `IdentitySafeExposureReplayGuard` construction. A coherently resealed packet cannot substitute a different ledger if the caller holds the expected identity.
2. `next_exposure_identity` / `authorize_batch_with_identity` bind current guard state, exact ledger/materialization/packing identities, claim sequence, ordered interval claims and exact nonignored target count; strict non-bool integer validation, segment bounds, replay/overlap and total budget checks fail closed.
3. `validate_exposure_plan_preflight` externally binds ledger + plan identities, proves exact total requested causal targets and one-pass capacity, rejects replay/overlap/unknown segments/type aliases/oversubscription and explicitly does **not** authorize training.
4. `ordered_next_exposure_identity` / `authorize_ordered_batch` bind the exact plan batch, worker/shard/order, current guard state and base next-exposure identity before state mutation.
5. `ordered_exposure_resume` reconstructs and validates the exact consumed plan prefix, checkpoint/trainer bindings, guard state identity, sequence/counts and next ordered identity. Fresh-process resume therefore cannot lawfully skip/reorder/replay a positive-loss interval merely by self-sealing a state packet.
6. Raw `ExposureReplayGuard` alone is lower-level and does not carry the whole external plan-prefix authority, but the canonical ordered/resume wrapper does; this is the same reason #1260's earlier raw-base suspicion was correctly falsified rather than patched.
7. #802's existing `tools/run_single_gpu_pilot.py` is explicitly `CONTROLLED_SYNTHETIC_MECHANICS_ONLY`; it is not a lawful corpus/learned-20M training carrier, so its lack of D04 corpus ledger calls is not an exposure-authority bypass. The real bounded learned-20M adapter is precisely the still-missing Product seam actively owned by #1248, whose acceptance criteria already require delegation to canonical D04 exposure/order guards before any real update.

NEGATIVE TEST REASONING covered at the external boundary:
- packet-authored coherent ledger substitution -> rejected by externally expected ledger SHA;
- unknown/missing keys / bool-as-int aliases -> rejected by exact-schema / strict integer guards;
- segment/budget/capacity drift -> rejected preflight/guard construction;
- replay/overlap/out-of-bounds -> rejected before commit;
- plan reorder/skip -> expected ordered identity / claim sequence mismatch;
- resumed prefix substitution -> prefix reconstruction + checkpoint/trainer/plan/ledger binding mismatch;
- oversubscription -> consumed + requested > authorized budget rejected.

SCIENTIFIC VERDICT:
`NO_NEW_D04_BLOCKER` for current-main canonical primitives. Do not reopen #1260's falsified raw-base repair and do not create a speculative D04 PR. The actual next risk is integration: the future #1248 bounded corpus-training adapter must *consume* these externally bound D04 primitives at the last loss-bearing boundary before optimizer update; it must not compute/accept a self-authored replacement authority.

TRUTH BOUNDARY remains unchanged:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
OPTIMIZER_UPDATES_EXECUTED=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

No Product file changed; no branch/PR created. Releasing this exact read-only audit semantic now and rotating SWARM-1280 to the next unowned critical blocker after a fresh live-state refresh.
